File: gpt6/python/probe.py
"""Deterministic numerical probe used only by the cross-language comparison."""
import json, sys
from engine import rank, information_gain
x = json.load(sys.stdin)
print(json.dumps({'ranking':rank(x['state'],x['tasks']),
                  'gains':[information_gain(*v) for v in x['cases']]}))

File: gpt6/typescript/probe.ts
/** Deterministic numerical probe for the cross-language comparison. */
import {readFileSync} from 'node:fs';
import {object,rank,informationGain,number} from './engine.ts';
const x=object(JSON.parse(readFileSync(0,'utf8')));
if(!Array.isArray(x.cases))throw new Error('Expected cases');
const gains=x.cases.map(v=>{
  if(!Array.isArray(v)||v.length!==3)throw new Error('Expected triple');
  return informationGain(number(v[0],'p'),number(v[1],'s'),number(v[2],'f'));
});
console.log(JSON.stringify({ranking:rank(x.state,x.tasks),gains}));
